.github/workflows/docker-build-push.yml

```diff
@@ -21,20 +31,63 @@ jobs:
             type=ref,event=tag
 
       - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1.7.0
+        uses: docker/setup-buildx-action@v2.0.0
 
       - name: Login to DockerHub
         if: github.event_name != 'pull_request'
-        uses: docker/login-action@v1.14.1
+        uses: docker/login-action@v2.0.0
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
 
       - name: Build image and push to Docker Hub
-        uses: docker/build-push-action@v2.10.0
+        uses: docker/build-push-action@v3.0.0
         with:
           context: .
           file: docker/Dockerfile
           push: ${{ github.event_name != 'pull_request' }}
           tags: ${{ steps.docker_metadata.outputs.tags }}
           labels: ${{ steps.docker_metadata.outputs.labels }}
+
+  deploy_to_gce:
+    name: Deploy the image to a GCE VM
+    runs-on: ubuntu-latest
+    needs: push_to_registry
+
+    # Add "id-token" with the intended permissions.
+    permissions:
+      contents: "read"
+      id-token: "write"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # Authentication via credentials json
+      - id: "auth"
+        uses: "google-github-actions/auth@v0"
+        with:
+          credentials_json: "${{ secrets.GCE_SA_KEY }}"
+
+      # Setup gcloud CLI
+      - name: Set up Cloud SDK
+        uses: google-github-actions/setup-gcloud@v0
+
+      # Make sure the VM is up
+      - name: Start the deploy-pudl-vm
+        run: gcloud compute instances start "$GCE_INSTANCE" --zone="$GCE_INSTANCE_ZONE"
```

Comment:
Does it matter that this takes a minute to complete? Does this action not return until the instance is ready to go? Does the `deploy` action need to explicitly depend on this action?

Reply:
Eventually, the container will shut the VM down so we don't pay for the VM when we aren't running a build. The github action needs to start the VM again before updating the container. Starting the VM and updating the container takes about a minute.

Reply:
Maybe we're talking past each other here.

Do the "steps" all happen in order, and wait until they complete before moving on to the next step? But then you have to explicitly setup dependencies with the `needs:` section between jobs?

Reply:
Oh my bad. Steps are executed in order. Jobs are not which is why I added the needs section between jobs. The `compute instances start` command returns when the VM has been started. This means the deploy step won't happen until the VM is up.